tests/waveplate_stub_test.py

```diff
@@ -0,0 +1,108 @@
+from pnpq.devices.waveplate_stub import WaveplateStub
+from pnpq.errors import (
+    DeviceDisconnectedError,
+    WaveplateInvalidDegreeError,
+    WaveplateInvalidMotorChannelError,
+    WaveplateInvalidStepsError,
+)
+import pytest
+
+def test_access_without_connection():
```

Comment:
Every method that should throw this error should be tested.

How do we do this quickly and easily?

Let's use a [parameterized test](https://docs.pytest.org/en/8.0.x/how-to/parametrize.html).

More generally, this pattern is called data-driven testing.

Using `eval` like in the pytest documentation is kind of scary (`eval` should be approached with extreme caution) but it's also the simplest way to accomplish this, so let's give it a try.